pkg/tools/grpcutils/url.go

```diff
@@ -93,3 +93,15 @@ func TargetToNetAddr(target string) (network, addr string) {
 
 	return network, target
 }
+
+// CheckURLFree returns check for is given url is free for Listen
```

Comment:
I don't think that this support any url, as I can see only tcp

Reply:
Fixed to support any URL.